Packs/Campaign/Scripts/script-ShowCampaignHighestSeverity.yml

```
@@ -3,48 +3,63 @@ commonfields:
-  MEDIUM_SEVERITY_COLOR = "rgb(209,125,0)"  #  Orange
+  MEDIUM_SEVERITY_COLOR = "rgb(209,125,0)"  # Orange
   HIGH_SEVERITY_COLOR = "rgb(209,60,60)"  # Red
   CRITICAL_SEVERITY_COLOR = "rgb(143,0,14)"  # Dark Red
 
-  # Variable initialization:
-  html = ""
-  severities = ""
-
-  try:
-      # Getting incident context:
-      incident_id = demisto.incidents()[0].get('id', {})
-      context = demisto.executeCommand("getContext", {'id': incident_id})
-      incidents = demisto.get(context[0], "Contents.context.EmailCampaign.incidents")
-      severities = [incident.get("severity") for incident in incidents]
-      current_severity = demisto.incidents()[0].get('severity', {})
-      severities.append(current_severity)
-      # Determine color:
-      if 4 in severities:
-          html = f"<div style='text-align:center; font-size:17px; padding: 15px;'> Highest Severity</br> <div style='font-size:32px; color:{CRITICAL_SEVERITY_COLOR};'> Critical </div></div>"
-      elif 3 in severities:
-          html = f"<div style='text-align:center; font-size:17px; padding: 15px;'> Highest Severity</br> <div style='font-size:32px; color:{HIGH_SEVERITY_COLOR};'> High </div></div>"
-      elif 2 in severities:
-          html = f"<div style='text-align:center; font-size:17px; padding: 15px;'> Highest Severity</br> <div style='font-size:32px; color:{MEDIUM_SEVERITY_COLOR};'> Medium </div></div>"
-      elif 1 in severities:
-          html = f"<div style='text-align:center; font-size:17px; padding: 15px;'> Highest Severity</br> <div style='font-size:32px; color:{LOW_SEVERITY_COLOR};'> Low </div></div>"
-      elif 0.5 in severities:
-          html =f"<div style='text-align:center; font-size:17px; padding: 15px;'> Highest Severity</br> <div style='font-size:32px; color:{INFORMATIONAL_SEVERITY_COLOR};'> Informational </div></div>"
-      else:
-          html =f"<div style='text-align:center; font-size:17px; padding: 15px;'> Highest Severity</br> <div style='font-size:32px; color:{INFORMATIONAL_SEVERITY_COLOR};'> Unknown </div></div>"
-  except Exception as ex:
-      html = f"<div style='text-align:center; padding: 20px;'> <div> No severity </div>"
-
-
-
-  # Return the data to the layout:
-  demisto.results({
-  'ContentsFormat': EntryFormat.HTML,
-  'Type': EntryType.NOTE,
-  'Contents': html
-  })
+  COLORS_AND_NAMES = {
+      0: {'color': INFORMATIONAL_SEVERITY_COLOR, 'dsc': 'Unknown'},
+      0.5: {'color': INFORMATIONAL_SEVERITY_COLOR, 'dsc': 'Informational'},
+      1: {'color': LOW_SEVERITY_COLOR, 'dsc': 'Low'},
+      2: {'color': MEDIUM_SEVERITY_COLOR, 'dsc': 'Medium'},
+      3: {'color': HIGH_SEVERITY_COLOR, 'dsc': 'High'},
+      4: {'color': CRITICAL_SEVERITY_COLOR, 'dsc': 'Critical'},
+  }
+
+
+  def get_incident_severity(incident_id):
+      data = demisto.executeCommand("getIncidents", {'id': incident_id})
+      try:
+          for key_or_index in [0, 'Contents', 'data', 0, 'severity']:
+              data = data[key_or_index]
```

try to convert the loop to dict_safe_get
```suggestion
      data = execute_command("getIncidents", {'id': incident_id})
      try:
          for key_or_index in ['data', 0, 'severity']:
              data = data[key_or_index]
```